pkg/components/state/pluggable.go

```diff
@@ -321,6 +331,46 @@ func (ss *grpcStateStore) Multi(ctx context.Context, request *state.Transactiona
 	return err
 }
 
+// MultiMaxSize returns the maximum number of operations allowed in a transactional request.
+func (ss *grpcStateStore) MultiMaxSize() int {
+	ss.lock.RLock()
+	multiMaxSize := ss.multiMaxSize
+	ss.lock.RUnlock()
+
+	if multiMaxSize != nil {
+		return *multiMaxSize
+	}
+
+	// In a worst case scenario, two or more callers may get to this point,
+	// meaning that we unnecessarily call MultiMaxSize multiple ties,
+	// however this is functionally not a problem and the cost is minimal.
+	ss.lock.Lock()
+	defer ss.lock.Unlock()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	resp, err := ss.Client.MultiMaxSize(ctx, new(proto.MultiMaxSizeRequest))
+	if err != nil {
+		log.Error("failed to get multi max size from state store", err)
+		ss.multiMaxSize = ptr.Of(-1)
+		return *ss.multiMaxSize
+	}
+
+	// If the pluggable component is on a 64bit system and the dapr runtime is on a 32bit system,
+	// the response will be larger than the maximum int32 value.
+	// In this case, we set the max size to the maximum possible value for a 32bit system.
+	is32bitSystem := unsafe.Sizeof(int(0)) == 4
```

Comment:
Do we need this temp value?

Reply:
Considering this is a local variable that's allocated on the stack, I think the extra readability is worth it, because it's making the code base more accessible to contributors from all levels. Happy to change it if that's not something that's very relevant at this point though.